### PR TITLE
Split start_nodes script ssh commands to individual scripts

### PR DESCRIPTION
**Pull request by pgarg66**

### CODE REVIEW

**Comment by pgarg66** [comment]
CI

File: multinode-demo/remote_leader.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
```
Comment (mvines) [inline multinode-demo/remote_leader.sh]: I like `#!/bin/bash -e` a lot :)

**Comment by pgarg66** [comment]
CI

File: multinode-demo/remote_leader.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
```
Comment (mvines) [inline multinode-demo/remote_leader.sh]: can we add a `[[ -n $FORCE ]] || exit` to this script, and the remote_validator.sh script.  This is not something we want a user to run by accident on their dev machine

File: multinode-demo/remote_leader.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+pkill -9 solana-
+
+# Run setup
+PATH="$HOME"/.cargo/bin:"$PATH"
+
+./fetch-perf-libs.sh
+
+USE_INSTALL=1 ./multinode-demo/setup.sh -p
+USE_INSTALL=1 SOLANA_CUDA=1 ./multinode-demo/leader.sh >leader.log 2>&1
+USE_INSTALL=1 ./multinode-demo/drone.sh >drone.log 2>&1
```
Comment (mvines) [inline multinode-demo/remote_leader.sh]: looks like the drone won't run here, leader isn't backgrounded

File: multinode-demo/remote_leader.sh (hunk 2 above)
Comment (pgarg66) [reply multinode-demo/remote_leader.sh]: Yes. I was wondering how to prevent it. This will ensure it.

File: multinode-demo/remote_leader.sh (hunk 1 above)
Comment (pgarg66) [reply multinode-demo/remote_leader.sh]: I'll change it.

File: multinode-demo/remote_leader.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+mkdir -p ~/.ssh ~/solana ~/.cargo/bin
```
Comment (rob-solana) [inline multinode-demo/remote_leader.sh]: replicode with remote_validator.sh

how about something that fetches our dependencies, instead?

Comment (pgarg66) [reply multinode-demo/remote_leader.sh]: Validator is fetching most dependencies from leader node ($1 is leader IP here).
One option to remove replicode is to split the overlap portion to a separate (remote_node.sh?) script. Are you thinking something beyond that?

File: multinode-demo/remote_leader.sh [hunk 5]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash -e
+
+[[ -n $FORCE ]] || exit
+
+mkdir -p ~/.ssh ~/solana ~/.cargo/bin
+sudo apt-get --assume-yes install rsync libssl-dev
+chmod 600 ~/.ssh/authorized_keys ~/.ssh/id_rsa
+
+# Run setup
```
Comment (mvines) [inline multinode-demo/remote_leader.sh]: stale comment?

File: multinode-demo/start_nodes.sh [hunk 6]
```diff
@@ -29,90 +29,76 @@ SECONDS=0
...
 common_setup() {
   ip_addr=$1
 
-  ssh -n -f "$remote_user@$ip_addr" 'mkdir -p ~/.ssh ~/solana ~/.cargo/bin'
-
   # Killing sshguard for now. TODO: Find a better solution
   # sshguard is blacklisting IP address after ssh-keyscan and ssh login attempts
-  ssh -n -f "$remote_user@$ip_addr" "sudo service sshguard stop"
-  ssh -n -f "$remote_user@$ip_addr" 'sudo apt-get --assume-yes install rsync libssl-dev'
+  ssh -n -f "$remote_user@$ip_addr" "sudo service sshguard stop" >log/"$ip_addr".log
```
Comment (mvines) [inline multinode-demo/start_nodes.sh]: Let's roll this into a single `ssh` session with:
```bash
ssh -n -f "$remote_user@$ip_addr" "\
  set -ex; \
  sudo service sshguard stop; \
  sudo apt-get --assume-yes install rsync libssl-dev; \
  pkill -9 leader; \
  pkill -9 validator; \
  pkill -9 solana- \
" >log/"$ip_addr".log
```

File: multinode-demo/remote_leader.sh (hunk 5 above)
Comment (pgarg66) [reply multinode-demo/remote_leader.sh]: Actually, useless comment at the wrong spot. Should be few lines down.

**Comment by garious** [comment]
Any reason this PR hasn't been merged yet?

**Comment by pgarg66** [comment]
I am not sure. Are we supposed to merge on our own, once all checks and reviews are clear? I was relying on you/mvines to merge stuff.

**Comment by mvines** [comment]
Nope, you should totally merge yourself once you have review

**Comment by garious** [comment]
Agreed. Merge away. Use PRs to improve cross-team communication.

**Pull request merged**